src/picongpu/include/plugins/particleCalorimeter/ParticleCalorimeter.hpp

```diff
@@ -0,0 +1,562 @@
+/**
+ * Copyright 2016 Heiko Burau
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "ParticleCalorimeterFunctors.hpp"
+#include "ParticleCalorimeter.kernel"
+
+#include "traits/PICToSplash.hpp"
+#include "plugins/ISimulationPlugin.hpp"
+#include "cuSTL/container/DeviceBuffer.hpp"
+#include "cuSTL/container/HostBuffer.hpp"
+#include "cuSTL/algorithm/kernel/Foreach.hpp"
+#include "cuSTL/cursor/MultiIndexCursor.hpp"
+#include "cuSTL/algorithm/mpi/Reduce.hpp"
+#include "cuSTL/algorithm/host/Foreach.hpp"
+#include "particles/policies/ExchangeParticles.hpp"
+#include "math/Vector.hpp"
+#include "algorithms/math.hpp"
+#include <boost/shared_ptr.hpp>
+
+/* libSplash data output */
+#include <splash/splash.h>
+#include <boost/filesystem.hpp>
+#include <string>
+#include <iostream>
+#include <fstream>
+#include <stdlib.h>
+
+namespace picongpu
+{
+using namespace PMacc;
+
+namespace po = boost::program_options;
+
+
+/** Virtual particle calorimeter plugin.
+ *
+ * (virtually) propagates and collects particles to infinite distance.
+ *
+ */
+template<class ParticlesType>
+class ParticleCalorimeter : public ISimulationPlugin
+{
+private:
+    std::string name;
+    std::string prefix;
+    std::string foldername;
+    uint32_t notifyPeriod;
+    MappingDesc* cellDescription;
+    std::ofstream outFile;
+    const std::string leftParticlesDatasetName;
+
+    uint32_t numBinsYaw;
+    uint32_t numBinsPitch;
+    uint32_t numBinsEnergy;
+    float_X minEnergy;
+    float_X maxEnergy;
+    bool logScale;
+    float_X openingYaw;
```

Comment:
please add `...Deg` to show that these are angles in degree not in radiant (as anywhere else)